suse/distributionscanner.go

```diff
@@ -111,11 +88,51 @@ func (ds *DistributionScanner) Scan(ctx context.Context, l *claircore.Layer) ([]
 // distribution if it exists.
 //
 // separated into its own method to aid testing.
-func (ds *DistributionScanner) parse(buff *bytes.Buffer) *claircore.Distribution {
-	for _, ur := range suseRegexes {
-		if ur.regexp.Match(buff.Bytes()) {
-			return releaseToDist(ur.release)
+func (ds *DistributionScanner) parse(ctx context.Context, buff *bytes.Buffer) *claircore.Distribution {
+	kv, err := osrelease.Parse(ctx, buff)
+	if err != nil {
+		zlog.Warn(ctx).Err(err).Msg("malformed os-release file")
+	}
+	cpeName, cpeOK := kv["CPE_NAME"]
+	if !cpeOK {
+		return nil
+	}
+	// Instead of regexing through, we can grab the CPE.
+	c, err := cpe.Unbind(cpeName)
+	if err != nil {
+		zlog.Warn(ctx).Err(err).Msg("could not unbind CPE")
+		return nil
+	}
+	ver, verOK := kv["VERSION_ID"]
+	if !verOK {
+		return nil
+	}
+
+	d, err := cpeToDist(c, ver)
+	if err != nil {
+		zlog.Warn(ctx).Err(err).Msg("error converting cpe to distribution")
+		return nil
+	}
+
+	return d
+}
+
+func cpeToDist(r cpe.WFN, ver string) (*claircore.Distribution, error) {
+	if vendor, err := cpe.NewValue("opensuse"); err == nil && r.Attr[cpe.Vendor] == vendor {
+		if prod, err := cpe.NewValue("leap"); err == nil && r.Attr[cpe.Product] == prod {
+			return mkLeapDist(r.String(), ver), nil
+		}
+	}
+	if vendor, err := cpe.NewValue("suse"); err == nil && r.Attr[cpe.Vendor] == vendor {
+		if prod, err := cpe.NewValue("sles"); err == nil && r.Attr[cpe.Product] == prod {
+			// Canonicalize the version to the major.
+			v, err := semver.NewVersion(ver)
```

Comment:
should we just use the version from the CPE?

Reply:
Yeah, that'd probably be cleaner